contents/swift/transition-to-swift.md

```diff
@@ -0,0 +1,340 @@
+# Swift
+
+Authors: [Ch'ng Ming Shin](https://github.com/ablyx/cs3281-website/blob/mingshin-week6-progress/students/AY1617S2/ch'ngMingShin/Ch'ngMingShin-Resume.md)
+
+# Overview
+
+Introduced in 2014, Swift is a fast, modern, safe and interactive programming language. If you have ever thought of building an app for your MacBook or iPhone (or any other Apple Product), Swift is the language for you. With its syntactic sugar, it is much more readable compared to it's "predecessor" Objective-C(ObjC). Together with its Playground feature, you can run your code easily, making Swift easy to learn.
+
+<!--
+If you have decided to build an app for your MacBook or iPhone (or any other Apple Product) or transitioning from another language to Swift, I am here to make your life easier. The common solutions to "How to learn Swift" from Quora and StackOverflow are to
+
+1. Dive into a project
+
+2. Read the official documentation
+https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/index.html
+
+
+Those are great suggestions but diving into a new project with no prior experience to Swift makes it an ardous trial of google searches of how to implement stuff. And reading documentation, while really helpful, might be overkill and boring as there might be many concepts that you already know.
+
+To ease this pain, I am providing a 3rd solution where I try to summarise the more common and important aspects of Swift that are generally different from other languages.
+-->
+
+# Getting Started
+
+## Good readings on Swift:
+- Design of Swift, why the change from obj-C:
+http://swiftjester.org/papers/whats-different-about-swift.html
+
+
+- Main Swift Concepts (less indepth, wide range of topics, long explanations):
+https://guides.codepath.com/ios/Swift-Basics
+
+- Basics of Apple Swift in playground:
+https://github.com/Codeido/swift-basics
+
+- Guide to Xcode(IDE for Swift):
+http://codewithchris.com/xcode-tutorial/
+
+- Another compilation of links:
+http://www.whoishostingthis.com/resources/swift/
+
+## Cheatsheets:
+- One page cheatsheet:
+https://koenig-media.raywenderlich.com/uploads/2014/06/RW-Swift-Cheatsheet-0_7.pdf
+
+- Multiple pages of cheatsheet (mainly code examples with comments):
+https://mhm5000.gitbooks.io/swift-cheat-sheet/content/index.html
+
+
+
+
+# Swift Basics
+
+## [The Absolute Basics](https://developer.apple.com/library/content/documentation/Swift/Conceptual/Swift_Programming_Language/TheBasics.html)
```

Comment:
It might be confusing to put links to the headline `##`. Maybe you want to explicitly write out the link. Like `[Read me](http://example.com)` (This is a bad exmaple. :P)
Or you can mention that the overall structure for each concept is first `reading link` and then `things to note`.